lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunication.cpp

```diff
@@ -1316,6 +1316,7 @@ std::string GDBRemoteCommunication::ExpandRLE(std::string packet) {
     } else {
       decoded.push_back(*c);
     }
+    c++;
```

Comment:
How does this prevent an out of bounds access? If I understand the problem correctly, the right fix to check for an almost-end iterator when doing the RLE expansion (change the condition to something like `if (*c == '*' && std::next(c) != packet.end())`, just less clumsy).